GenderPayGap.WebUI/Controllers/Admin/AdminOrganisationSectorController.cs

```diff
@@ -128,6 +205,41 @@ private void AddNewOrganisationPublicSectorType(Organisation organisation, int p
 
             organisation.LatestPublicSectorType = newOrganisationPublicSectorType;
         }
+        
+        private void UpdateAdminChangeSectorViewModelFromOrganisation(AdminChangeSectorViewModel viewModel, long organisationId)
+        {
+            viewModel.Organisation = dataRepository.Get<Organisation>(organisationId);
+        }
+        
+        private void ValidateAdminChangeSectorViewModel(AdminChangeSectorViewModel viewModel)
+        {
+            viewModel.ParseAndValidateParameters(Request, m => m.NewSector);
+            viewModel.ParseAndValidateParameters(Request, m => m.Reason);
+        }
+        
+        private void ChangeSector(AdminChangeSectorViewModel viewModel, long organisationId)
+        {
+            var organisation = dataRepository.Get<Organisation>(organisationId);
+
+            SectorTypes previousSector = organisation.SectorType;
+            SectorTypes newSector = viewModel.NewSector.Value == NewSectorTypes.Private ? SectorTypes.Private : SectorTypes.Public;
+
+            // Update the sector
+            organisation.SectorType = newSector;
+
+            // Remove SIC codes when company changes between sectors
+            organisation.OrganisationSicCodes.Clear();
+
+            dataRepository.SaveChangesAsync().Wait();

```

Comment:
We should also change all the...
* `OrganisationScope.SnapshotDate`
* `Return.AccountingDate`

... to match the new sector.

You can use the method `GetAccountingStartDate` in `GenderPayGap.Core\Classes\Extensions.cs` to get the right value